pkg/adapter/datasource.go

```diff
@@ -105,7 +106,16 @@ func (d *Datasource) GetPage(ctx context.Context, request *Request) (*Response,
 	// SCAFFOLDING #17 - pkg/adapter/datasource.go: Add any headers required to communicate with the SoR APIs.
 	// Add headers to the request, if any.
 	// req.Header.Add("Accept", "application/json")
-	// req.Header.Add("Authorization", "Bearer Token")
+
+	if request.Token == "" {
+		// Basic Authentication
+		auth := request.Username + ":" + request.Password
+		encodedAuth := base64.StdEncoding.EncodeToString([]byte(auth))
+		req.Header.Add("Authorization", "Basic "+encodedAuth)
```

Comment:
Let's not create intermediate variables that'll only be used once.

```suggestion
		req.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(auth)))
```

Reply:
Done